tests/attributes/__init__.py

```diff
@@ -0,0 +1,103 @@
+import json
+import logging
+from typing import Callable, NoReturn, Dict, List, Optional
+
+import pytest
+from webdriver_recorder.browser import BrowserRecorder
+
+from tests.helpers import WebTestUtils
+
+
+class AttributeReleaseTestBase:
+    # These are just type declarations.
+    # Do not set values on these fields!
+    fresh_browser: BrowserRecorder
+    login: str
+    utils: WebTestUtils
+    sp_shib_url: Callable[..., str]
+    test_env: str
+    enter_duo_passcode: Callable[..., NoReturn]
+    log_in_netid: Callable[..., NoReturn]
+    idp_env: str
+
+    @pytest.fixture(autouse=True)
+    def initialize_base(
+            self,
+            netid,
+            utils,
+            sp_shib_url,
+            test_env,
+            log_in_netid,
+            enter_duo_passcode,
+    ):
+        self.login = netid
+        self.utils = utils
+        self.sp_shib_url = sp_shib_url
+        self.log_in_netid = log_in_netid
+        self.enter_duo_passcode = enter_duo_passcode
+
+        self.idp_env = ''
+        if test_env == "eval":
+            self.idp_env = ":eval"
+
+    def _parse_line(self, line):
+        parts = line.split('= ', maxsplit=1)
+        if len(parts) == 1:
+            parts.append('')
+        return tuple(map(lambda p: p.strip(), parts))
```

Comment:
great use of `map`. i often opt for a dictcomp but i think this is more readable because of the tuple casting.